web/viewer.css

```diff
@@ -1372,129 +1271,110 @@ dialog :link {
 }
 
 #findbar {
-  background-color: var(--toolbar-bg-color);
-  cursor: default;
-  font: message-box;
-  font-size: 12px;
+  --input-horizontal-padding: 4px;
+  --findbar-padding: 2px;
+
+  width: max-content;
+  max-width: 90vw;
+  min-height: var(--toolbar-height);
   height: auto;
-  inset-inline-start: 64px;
-  line-height: 14px;
-  margin: 4px 2px;
-  min-width: 300px;
-  padding: 0 4px;
   position: absolute;
-  text-align: left;
-  top: var(--toolbar-height);
   z-index: 30000;
+  cursor: default;
+  padding: 0;
+  min-width: 300px;
+  background-color: var(--toolbar-bg-color);
+  box-sizing: border-box;
+  flex-wrap: wrap;
+  justify-content: flex-start;
 
-  * {
-    float: var(--inline-start);
-    position: relative;
-  }
-
-  > div {
+  > * {
     height: var(--toolbar-height);
+    padding: var(--findbar-padding);
   }
 
-  :is(button, input) {
-    font: message-box;
-    outline: none;
-  }
+  #findInputContainer {
+    margin-inline-start: 2px;
 
-  input {
-    &[type="checkbox"] {
-      pointer-events: none;
-
-      &:checked + .toolbarLabel {
-        background-color: var(--toggled-btn-bg-color) !important;
-        color: var(--toggled-btn-color);
-      }
+    #findPreviousButton::before {
+      mask-image: var(--findbarButton-previous-icon);
     }
-  }
-
-  label {
-    user-select: none;
-  }
 
-  :is(label:hover, input:focus-visible + label) {
-    background-color: var(--button-hover-color);
-    color: var(--toggled-btn-color);
-  }
-
-  #findbarInputContainer {
-    margin-inline-end: 4px;
+    #findNextButton::before {
+      mask-image: var(--findbarButton-next-icon);
+    }
 
     #findInput {
       width: 200px;
+      padding: 5px var(--input-horizontal-padding);
 
       /*#if !MOZCENTRAL*/
       &::-webkit-input-placeholder {
         color: rgb(191 191 191);
       }
-      /*#endif*/
 
+      /*#endif*/
       &::placeholder {
         font-style: normal;
       }
 
       .loadingInput:has(> &[data-status="pending"])::after {
-        display: block;
+        display: inline;
         visibility: visible;
+        inset-inline-end: calc(var(--input-horizontal-padding) + 1px);
       }
 
       &[data-status="notFound"] {
         background-color: rgb(255 102 102);
       }
     }
-
-    #findPreviousButton::before {
-      mask-image: var(--findbarButton-previous-icon);
-    }
-
-    #findNextButton::before {
-      mask-image: var(--findbarButton-next-icon);
-    }
   }
 
   #findbarMessageContainer {
+    display: none;
+    gap: 4px;
+
     #findResultsCount {
       background-color: rgb(217 217 217);
       color: rgb(82 82 82);
-      margin: 5px;
-      padding: 4px 5px;
-      text-align: center;
+      padding-block: 4px;
+
+      &:empty {
+        display: none;
+      }
+    }
+
+    &:has(> :is(#findResultsCount, #findMsg):not(:empty)) {
+      display: inline flex;
     }
```

Comment:
Nit: Rather than placing this rule "between" the `#findResultsCount` and `#findMsg` blocks, can we please move it either before or after instead?